uitools/common/src/PopupAttachmentItem.cpp

```diff
@@ -136,8 +135,16 @@ void PopupAttachmentItem::downloadAttachment()
 {
   m_fetchingAttachment = true;
   emit popupAttachmentItemChanged();
-  m_popupAttachment->attachment()->fetchDataAsync().then([this] (const QByteArray&)
+  m_popupAttachment->attachment()->fetchDataAsync().then([this] (const QByteArray& attachmentData)
   {
+    if (attachmentData.isEmpty())
+    {
+      m_fetchingAttachment = false;
+      emit popupAttachmentItemChanged();
+      return;
+    }
+
+    emit attachmentDataFetched(attachmentData, name());
```

Comment:
Ditto - what needs this data? If it is necessary for the popup to function, then we need to keep it.